components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -1404,16 +1412,32 @@ public void revokeAccessTokensInBatch(String[] tokens, boolean isHashedToken) th
                     } else {
                         ps.setString(3, getHashingPersistenceProcessor().getProcessedAccessTokenIdentifier(token));
                     }
+                    ps.setInt(4, IdentityTenantUtil.getLoginTenantId());
                     ps.addBatch();
-                    oldTokens.add(getHashingPersistenceProcessor().getProcessedAccessTokenIdentifier(token));
                 }
-                ps.executeBatch();
+                int[] countArray = ps.executeBatch();
                 IdentityDatabaseUtil.commitTransaction(connection);
-                // To revoke request objects which have persisted against the access token.
-                OAuth2TokenUtil.postUpdateAccessTokens(Arrays.asList(tokens), OAuthConstants.TokenStates.
-                        TOKEN_STATE_REVOKED);
-                if (isTokenCleanupFeatureEnabled) {
-                    oldTokenCleanupObject.cleanupTokensInBatch(oldTokens, connection);
+
+                List<String> oldTokens = new ArrayList<>();
+                List<String> oldTokenHashes = new ArrayList<>();
+                for (int i = 0; i < tokens.length; i++) {
+                    if (countArray[i] > 0) {
+                        oldTokens.add(tokens[i]);
+                        if (isHashedToken) {
+                            oldTokenHashes.add(tokens[i]);
+                        } else {
+                            oldTokenHashes.add(getHashingPersistenceProcessor().getProcessedAccessTokenIdentifier(
+                                    tokens[i]));
+                        }
+                    }
+                }
```

Comment:
not clear on this change

Reply:
Resolved with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2131#discussion_r1336755958